logger.go

```diff
@@ -0,0 +1,55 @@
+package gondola
+
+import (
+	"context"
+	"log/slog"
+	"os"
+
+	"github.com/google/uuid"
+)
+
+// Logger is a logger.
+type Logger struct {
+	*slog.Logger
+}
+
+// NewLogger creates a logger.
+func NewLogger(level int) *Logger {
+	handler := TraceIDHandler{slog.NewJSONHandler(os.Stdout, &slog.HandlerOptions{
+		Level: slog.Level(level),
+	})}
+	logger := slog.New(handler)
+	return &Logger{
+		logger,
+	}
+}
+
+// WithTraceID adds a trace ID to the context.
+func WithTraceID(ctx context.Context) context.Context {
+	uuid, _ := uuid.NewRandom()
+	return context.WithValue(ctx, ctxTraceIDKey, uuid.String())
+}
+
+// GetTraceID returns a trace ID from the context.
+func GetTraceID(ctx context.Context) string {
+	tid, _ := ctx.Value(ctxTraceIDKey).(string)
+	return tid
+}
```

Comment:
I think the only way to manipulate the value of context is to make it a public method.
Any goods ideas?